### PR TITLE
MAINT: drop Python 3.9 and NumPy 1.22.x

### PR DESCRIPTION
**Pull request by rgommers**
We kept these a bit longer in order to have no changes in `1.13.x` other than building against numpy `2.0.0` pre-releases. It's time to drop these versions now though (see NEP 29 & SPEC 0).

### CODE REVIEW

**Review by h-vetinari — APPROVED** [review]
👍

**Comment by rgommers** [comment]
The one failure is unrelated, due to outdated `cibuildwheel` (see https://github.com/scipy/scipy/pull/20362#issuecomment-2030006848).

I see I missed one minor thing though, the job name `cp39 (meson) full, dev.py` needed a tweak. Will update that.

File: .github/workflows/windows.yml [hunk 1]
```diff
@@ -49,7 +49,7 @@ jobs:
 
       - name: pip-packages
         run: |
-          pip install numpy==1.22.4 cython pybind11 pythran meson ninja pytest pytest-xdist pytest-timeout pooch rich_click click doit pydevtool hypothesis "scipy-openblas32<=0.3.23.293.2"
+          pip install numpy==1.23.5 cython pybind11 pythran meson ninja pytest pytest-xdist pytest-timeout pooch rich_click click doit pydevtool hypothesis "scipy-openblas32<=0.3.23.293.2"
```
Comment (tylerjereddy) [inline .github/workflows/windows.yml:52]: I guess there's a `distutils` requirement missing on this from-source `pip` install of NumPy in the CI? Can't remember if i.e., disabling the build isolation and using a newer `setuptools` that vendors it (?) is one way to fix. But also not sure if worth it here.

Comment (rgommers) [reply .github/workflows/windows.yml]: The `==1.23.5` wasn't needed. The `cp310` job is oldest versions; the `cp312` can be with the latest numpy release. All green now.

**Comment by tylerjereddy** [comment]
Thanks Ralf.

I imagine folks may find 1 or 2 places where we can remove old code path guards, though a quick `git grep` didn't show anything immediately obvious.

**Pull request merged**